websites/T/TomsHardware/presence.ts

```diff
@@ -0,0 +1,96 @@
+  } else if (document.location.href.includes("best-picks")) {
+    if (pathname.split("/")[2]) {
+      presenceData.details = "Looking at best picks...";
+      presenceData.state = `The best picks for: '${pathname
+        .split("/")[3]
+        .replace(".html", "")
+        .replace("best-", "")}'`;
+    } else presenceData.details = "Scrolling through best picks...";
+  } else if (document.location.href.includes("coupons")) {
+    if (pathname.includes("?")) presenceData.details = "Looking at a coupon...";
+    else if (pathname.split("/")[2]) {
+      presenceData.details = "Looking at coupons...";
+      presenceData.state = `Coupons for: '${pathname
+        .split("/")[3]
+        .replace(".html", "")}'`;
+    } else presenceData.details = "Scrolling through coupons...";
+  } else if (document.location.href.includes("forums.")) {
+    if (pathname.includes("forums")) {
+      presenceData.state = `Category: '${pathname
+        .split("/")[2]
+        .split(".")[0]
+        .replace(".html", "")}'`;
+    } else if (pathname.includes("threads")) {
+      presenceData.state = `Category: '${pathname
+        .split("/")[2]
+        .split(".")[0]
+        .replace(".html", "")}'`;
+    } else if (pathname.includes("featured"))
+      presenceData.state = "Looking at featured threads...";
+    else if (pathname.includes("whats-new"))
+      presenceData.state = "Looking at new threads...";
+    else if (pathname.includes("members"))
+      presenceData.state = "Looking at members...";
+    else if (pathname.includes("help")) presenceData.state = "Getting help...";
+    else if (pathname.includes("search"))
+      presenceData.state = "Searching the forums...";
+    else if (pathname.includes("billboard"))
+      presenceData.state = "Looking at the billboard...";
+    else if (pathname.includes("register"))
+      presenceData.state = "Registering an account...";
+
+    presenceData.details = "Visiting the forums...";
+  } else if (document.location.href.includes("topics")) {
+    if (pathname.split("/")[3]) {
+      presenceData.state = `Subcategory: '${pathname
+        .split("/")[3]
+        .replace(".html", "")}'`;
+    }
+    presenceData.details = `Looking at the topic: ${pathname.split("/")[2]}`;
+  } else if (document.location.href.includes("feeds"))
+    presenceData.details = "Visiting the RSS page...";
+  else if (document.location.href.includes("search")) {
+    if (pathname.split("=")[1]) {
+      presenceData.state = `Searching Term: '${pathname
+        .split("=")[1]
+        .replace(".html", "")}'`;
+    }
+    presenceData.details = "Searching the site...";
+  } else if (pathname && pathname.length > 2) {
+    presenceData.details = "Browsing...";
+    presenceData.state = `On the page '${pathname.split("/")[1]}'`;
+  } else presenceData.details = "Visiting the front page...";
+
+  if (!presenceData.details) {
+    presence.setTrayTitle();
+    presence.setActivity();
+  } else presence.setActivity(presenceData);
```

Comment:
```suggestion
  presence.setActivity(presenceData);
```
I think you can just do this because a state or detail is always set using the last to `if`/`else` cases

Reply:
I got that part from the documentation [here](https://docs.premid.app/en/dev/presence) under the **Getting started** part though

Reply:
nvm read that wrong, I can do that, you're right